cdap-security-spi/src/main/java/co/cask/cdap/security/spi/authorization/TxCallable.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.security.spi.authorization;
+
+import co.cask.cdap.api.data.DatasetContext;
+import co.cask.cdap.api.dataset.Dataset;
+
+/**
+ * A callable that provides a {@link DatasetContext} which may be used to get access to and use datasets
+ * in a transaction and return a value.
+ * @param <RETURN_TYPE> return type of the {@link #call(DatasetContext)} method
+ */
+public interface TxCallable<RETURN_TYPE> {
```

Comment:
The type returned by the call method is called result type, not return type in the JDK Callable interface. I think we should follow the same (actually the type param is just V and it's javadoc refer it as the result type of the call method).